[FLINK-32852][JUnit5 Migration] Migrate the scheduler package of flink-runtime module to JUnit5

RocMarshal opened the pull request:
<!--
*Thank you very much for contributing to Apache Flink - we are happy that you want to help us improve Flink. To help the community review your contribution in the best possible way, please go through the checklist below, which will get the contribution into a shape in which it can be best reviewed.*

*Please understand that we do not do this to make contributions to Flink a hassle. In order to uphold a high standard of quality for code contributions, while at the same time managing a large number of contributions, we need contributors to prepare the contributions well, and give reviewers enough contextual information for the review. Please also understand that contributions that do not follow this guide will take longer to review and thus typically be picked up with lower priority by the community.*

## Contribution Checklist

  - Make sure that the pull request corresponds to a [JIRA issue](https://issues.apache.org/jira/projects/FLINK/issues). Exceptions are made for typos in JavaDoc or documentation files, which need no JIRA issue.
  
  - Name the pull request in the form "[FLINK-XXXX] [component] Title of the pull request", where *FLINK-XXXX* should be replaced by the actual issue number. Skip *component* if you are unsure about which is the best component.
  Typo fixes that have no associated JIRA issue should be named following this pattern: `[hotfix] [docs] Fix typo in event time introduction` or `[hotfix] [javadocs] Expand JavaDoc for PuncuatedWatermarkGenerator`.

  - Fill out the template below to describe the changes contributed by the pull request. That will give reviewers the context they need to do the review.
  
  - Make sure that the change passes the automated tests, i.e., `mvn clean verify` passes. You can set up Azure Pipelines CI to do that following [this guide](https://cwiki.apache.org/confluence/display/FLINK/Azure+Pipelines#AzurePipelines-Tutorial:SettingupAzurePipelinesforaforkoftheFlinkrepository).

  - Each pull request should address only one issue, not mix up code from multiple issues.
  
  - Each commit in the pull request has a meaningful commit message (including the JIRA id)

  - Once all items of the checklist are addressed, remove the above text and this checklist, leaving only the filled out template below.


**(The sections below can be removed for hotfixes of typos)**
-->

## What is the purpose of the change

[FLINK-32852][JUnit5 Migration] Migrate the scheduler package of flink-runtime module to JUnit5


## Brief change log

[FLINK-32852][JUnit5 Migration] Migrate the scheduler package of flink-runtime module to JUnit5


## Verifying this change


This change is already covered by existing tests, such as *(please describe tests)*.


## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

RocMarshal commented:
hi, @JingGe Could you help take a look if you had the free time ?
Thank you very much~

1996fanrui commented on flink-runtime/src/test/java/org/apache/flink/runtime/testutils/InternalMiniClusterExtension.java:
```diff
@@ -109,4 +117,24 @@ public Object resolveParameter(
         }
         throw new ParameterResolutionException("Unsupported parameter");
     }
+
+    @Override
+    public void beforeEach(ExtensionContext context) throws Exception {
+        miniClusterResource.before();
+    }
+
+    @Override
+    public void afterEach(ExtensionContext context) throws Exception {
+        miniClusterResource.after();
+    }
+
+    @Override
+    public void before(ExtensionContext context) throws Exception {
+        miniClusterResource.before();
+    }
+
+    @Override
+    public void after(ExtensionContext context) throws Exception {
+        miniClusterResource.after();
+    }
```
I saw the mini cluster resource didn't close for each test before. I guess one reason is performance.

Start a mini cluster needs some times, if we restart it frequently, test will take a long time. 

Is it necessary for this PR?

<img width="1043" alt="image" src="https://github.com/apache/flink/assets/38427477/4343a850-9472-48ea-b331-5e6fd94ae9de">

1996fanrui commented on flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/StateWithExecutionGraphTest.java:
```diff
@@ -120,9 +123,8 @@ public void testOnGloballyTerminalStateCalled() throws Exception {
+                .isCompletedWithValue(JobStatus.FINISHED);
     }
 
     @Test
```
All public can be removed.

1996fanrui commented on flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/adaptive/BackgroundTaskTest.java:
```diff
@@ -18,55 +18,52 @@
     }
 
     @Test
     public void testNormalCompletionOfBackgroundTask() {
```
All public can be removed.

1996fanrui commented on flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtilsTest.java:
```diff
@@ -182,21 +182,20 @@ private void triggerComputeNumOfSubpartitions(IntermediateResult result) {
             List<SlotSharingGroup> slotSharingGroups, List<MemorySize> networkMemory) {
 
-        assertEquals(slotSharingGroups.size(), networkMemory.size());
+        assertThat(networkMemory.size()).isEqualTo(slotSharingGroups.size());
```
hasSameSizeAs

1996fanrui commented on flink-runtime/src/test/java/org/apache/flink/runtime/scheduler/SsgNetworkMemoryCalculationUtilsTest.java:
```diff
@@ -270,16 +269,16 @@ private void createExecutionGraphAndEnrichNetworkMemory(
 
-        assertThat(slotSharingGroups.size(), is(3));
-        assertThat(parallelisms.size(), is(3));
+        assertThat(slotSharingGroups.size()).isEqualTo(3);
+        assertThat(parallelisms.size()).isEqualTo(3);
```
hasSize

1996fanrui reviewed (commented):
Some classes should be changed, but they are missed, so I commented here:

- Remove @ExtendWith(TestLoggerExtension.class)
  - EnforceMinimalIncreaseRescalingControllerTest
  - DefaultExecutionDeployerTest
  - StopWithSavepointTest
- Don't use `org.junit.Assert`
  - StateValidator
  - SchedulerTestingUtils
  - MockStateWithExecutionGraphContext
- public can be removed for test method
  - AdaptiveSchedulerTest
  - StateLocalitySlotAssignerTest
-  public can be removed for test class
  - you can search `public class`, I saw a lot of classes need to remove it.
- Update `isEqualTo(0)` to `isZero`
  - AdaptiveSchedulerTest
  - AllToAllBlockingResultInfoTest
 - Update `isEqualTo(1)` to `isOne`
   - ExecutionTimeBasedSlowTaskDetectorTest
   - SpeculativeExecutionTest
   - PointwiseBlockingResultInfoTest
   - DefaultVertexParallelismAndInputInfosDeciderTest
   - AdaptiveSchedulerTest
   - AllToAllBlockingResultInfoTest
 - hasSameSizeAs
   - DefaultVertexParallelismAndInputInfosDeciderTest (line 445)

RocMarshal commented:
> public can be removed for test method
AdaptiveSchedulerTest
StateLocalitySlotAssignerTest

Thank you very much for the patient-check and review.

There're some class referenced by other class. so I kept the `public` key-word for the related class and removed the key-word for some class.

As shown in the commit messages, I split the patches by commit.
I'd be appreciated with your next confirmation!

1996fanrui commented:
@flinkbot run azure

1996fanrui reviewed (approved):
Thanks @RocMarshal for the update!

LGTM
